src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -995,6 +996,9 @@ ReactDOMComponent.Mixin = {
       }
       if (propKey === STYLE) {
         if (nextProp) {
+          if (Array.isArray(nextProp)) {
```

Comment:
It seems like people keep adding complexity to this function, a little at a time, and it's getting longer and longer. And it doesn't seem to be covered by the test for this file, although I didn't read the whole test yet. 
After reading the comparable code in ReactNative I think this would be more maintainable if pulled out and refactored. Just didn't want to start down that path without some idea of whether it's worth it.